test/testdrive/linked-clusters.td

```diff
@@ -0,0 +1,220 @@
+# with the default size. (Unsafe mode only.)
+> CREATE SOURCE lg FROM LOAD GENERATOR COUNTER
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_public_lg
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                r1      ${arg.default-replica-size}
+mz_introspection       r1      1
+mz_system              r1      1
+materialize_public_lg  linked  ${arg.default-storage-size}
+> DROP SOURCE lg
+
+# Test that creating a source without a the REMOTE option creates a linked
+# cluster with no replicas. (Unsafe mode only.)
+> CREATE SOURCE lg FROM LOAD GENERATOR COUNTER WITH (REMOTE 'localhost:32323')
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_public_lg
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                r1      ${arg.default-replica-size}
+mz_introspection       r1      1
+mz_system              r1      1
+> DROP SOURCE lg
+
+# Test that creating sources with an explicit size creates linked clusters, each
+# with a replica named "linked" with the size specified in the source.
+> CREATE SOURCE lg1 FROM LOAD GENERATOR COUNTER WITH (SIZE = '1')
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_public_lg1
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                 r1      ${arg.default-replica-size}
+mz_introspection        r1      1
+mz_system               r1      1
+materialize_public_lg1  linked  1
+> CREATE SOURCE lg2 FROM LOAD GENERATOR COUNTER WITH (SIZE = '2')
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_public_lg1
+materialize_public_lg2
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                 r1      ${arg.default-replica-size}
+mz_introspection        r1      1
+mz_system               r1      1
+materialize_public_lg1  linked  1
+materialize_public_lg2  linked  2
+
+# Test that changing a source size results in the cluster being dropped and
+# recreated.
+> ALTER SOURCE lg2 SET (SIZE = '4')
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_lg2'
+materialize_public_lg2  linked  4
+
+# Test that a source can be converted to a REMOTE source and then back to a
+# managed source. (Unsafe mode only.)
+> ALTER SOURCE lg2 SET (REMOTE = 'localhost:32323')
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_lg2'
+> ALTER SOURCE lg2 SET (SIZE = '4')
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_lg2'
+materialize_public_lg2  linked  4
+> ALTER SOURCE lg2 RESET (SIZE)
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_lg2'
+materialize_public_lg2  linked  ${arg.default-storage-size}
+
+# Test that linked clusters cannot be dropped, nor can their replicas be
+# dropped, nor can new replicas be created.
+! DROP CLUSTER materialize_public_lg2
```

Comment:
I guess `DROP CLUSTER CASCADE ` should have worked here , but this may be outside of the scope of this PR.

Reply:
Ah, good point that that should be tested. Added. I think it is actually correct for `DROP ... CASCADE` not to work. Cluster links are very strong (and weird), and the idea is that you literally cannot manipulate the linked cluster at all. Not even via `DROP CASCADE`.

Reply:
Put another way: I think of a linked cluster like an index on a table that is force created and cannot be dropped. If you had such an index, it would be surprising for `DROP INDEX ... CASCADE` to result in the drop of the table.